### ISSUE TITLE
[firewalld 1.3.1] testsuite: 140 207 208 209 221 224 232 238 243 244 245 247 255 259 264 268 274 277 283 284 285 291 292 293 294 295 297 298 299 300 301 302 303 304 305 306 307 308 309 failed

### ISSUE THREAD

**Issue by mbiebl** [firewalld/firewalld#1112]
### What happened

Trying to run the test suite on Debian unstable/bookworm with kernel 6.1.0-7-amd64 failed

One repeating pattern in the failing tests is something like
```
-ACCEPT sctp -- 0.0.0.0/0 0.0.0.0/0
-ACCEPT icmpv6-- 0.0.0.0/0 0.0.0.0/0
+ACCEPT 132 -- 0.0.0.0/0 0.0.0.0/0
+ACCEPT 58 -- 0.0.0.0/0 0.0.0.0/0
```
symbolic names vs numbers

### What you expected to happen

Test suite to pass

### How to reproduce it (as minimally and precisely as possible)

run the test suite in a minimal Debian sid VM
In this specific case, I was using autopkgtest
```
autopkgtest-build-qemu sid /srv/chroot/autopkgtest-sid.img
autopkgtest firewalld -- qemu /srv/chroot/autopkgtest-sid.img
```

### Firewalld Version

1.3.1

### Firewalld Backend

nftables

### Linux distribution

Debian Sid 

### Linux kernel version

 6.1.0-7-amd64

### Other information

Full log is attached

[log.txt](https://github.com/firewalld/firewalld/files/11266081/log.txt)

**Comment by erig0** [comment]
This looks like an output change in `iptables-nft`. I'm not sure if that is an intentional change or not. I'll look into it, thanks for reporting!

**Comment by mbiebl** [comment]
I'd probably go with 2/, after all, firewalld is also built for older distros where you probably can't expect the latest iptables.

I wouldn't rely on the fixed behavior of iptables until at least such a version has reached the stable/LTS versions of major distros.

**Comment by mbiebl** [comment]
For completeness sake: Debian Sid currently has iptables 1.8.9-2 and stable 1.8.7-1

**Comment by mbiebl** [comment]
> > I'd probably go with 2/, after all, firewalld is also built for older distros where you probably can't expect the latest iptables.
> 
> Do you mean option 1?

ahem, yeah :-)

**Comment by erig0** [comment]
> > > I'd probably go with 2/, after all, firewalld is also built for older distros where you probably can't expect the latest iptables.
> > 
> > 
> > Do you mean option 1?
> 
> ahem, yeah :-)

See point about the reverse DNS lookups.

**Comment by erig0** [comment]
> > * will cause reverse DNS lookups (likely why we use it in the testsuite)
> 
> hm, that complicates things. It appears da8ecc62dd76 has been released as part of iptables 1.8.9. So Debian stable/bullseye would be one of the stable distros, where 2/ would fail

Right. I'm looking into using `itables -S`, but that would require updating _all_ the expected output in the testsuite.

**Comment by mbiebl** [comment]
As far as Debian is concerned, I could ask the iptables maintainer to provide a backport of 1.8.9 for bullseye
But see e.g. Ubuntu
https://packages.ubuntu.com/search?keywords=iptables&searchon=names&suite=all&section=all

**Comment by erig0** [comment]
> What about a piping the output of iptables through a custom script which translates symbolic names to numbers?

Yeah. That's an option. We already do a lot of munging of output in the testsuite.

https://github.com/firewalld/firewalld/blob/01287b32bafe5deb9a9cb5d5bf9d0e854eee4a4e/src/tests/functions.at#L399-L404

**Comment by erig0** [comment]
This is trivially reproduced with the container tests..

```
# make -C src/tests check-container-debian-sid TESTSUITEFLAGS="-j20"
```

**Comment by erig0** [comment]
iptables commit `6e41c2d8747b ("iptables: xshared: Ouptut '--' in the opt field in ipv6's fake mode")` added `--` to the ip6tables output. So I'm adding normalization for that as well.

**Comment by erig0** [comment]
@mbiebl , please give #1113 a try. It has a lot of changes to the testsuite, but should apply cleanly to v1.3.1.

**Comment by mbiebl** [comment]
> @mbiebl , please give #1113 a try. It has a lot of changes to the testsuite. It almost certainly won't apply clean to v1.3.1 due to other changes on the master branch that made a lot of testsuite changes.

thanks, will give it a try.
Btw, one option we didn't consider is asking the iptables maintainers if the (breaking) change is actually intended.
After all, it changes the output of iptables and I guess firewalld is not the only one parsing it.

**Comment by erig0** [comment]
> thanks, will give it a try

I'm working on a v1.3.2 release. So you can wait on that if you want.

**Comment by mbiebl** [comment]
Thanks, will wait then

**Comment by mbiebl** [comment]
```
309 tests were successful.
2 tests were skipped.
```

**Comment by mbiebl** [comment]
Apparently this upstream change was reverted (and backported into Debian), which now causes the test suite to fail again:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1067733
https://bugzilla.netfilter.org/show_bug.cgi?id=1729
https://ci.debian.net/packages/f/firewalld/testing/amd64/47753373/

**Comment by erig0** [comment]
AFAICS, the debian tests are failing not because of protocol numbers, but because of duplicate `--` in the output.

**Comment by mbiebl** [comment]
@erig0 true, but the only funtional changes between that happened to the Debian iptables package is
https://salsa.debian.org/pkg-netfilter-team/pkg-iptables/-/commit/2ed6c0b10b390ea1c1658226b00d5b85f98872a7

**Comment by erig0** [comment]
Hrm. Perhaps the sed expression in the firewalld testsuite needs work.

I may not get to it this week due to other urgent matters.

**Comment by mbiebl** [comment]
yeah, the output mangling was also my first guess.

**Comment by erig0** [comment]
This is looking like a bad `sed` expression. It runs fine on a different sed version. 🤷 

Debian Sid:
```
# echo "ACCEPT icmpv6 -- ::/0 ::/0" | sed -e 's/^\([-a-zA-Z0-9_]\+\)[ ]icmpv6/\1 58 /g' -e 's/^\([-a-zA-Z0-9_]\+\)[ ]\([^ ]*\)[ ][-]\?[-]\?[ ]\?/\1 \2 -- /g'
ACCEPT 58 -- -- ::/0 ::/0

root@a577aa2d3f86:/# sed --version
sed (GNU sed) 4.9
```

OS X:
```
% echo "ACCEPT icmpv6 -- ::/0 ::/0" | sed -e 's/^\([-a-zA-Z0-9_]\+\)[ ]icmpv6/\1 58 /g' -e 's/^\([-a-zA-Z0-9_]\+\)[ ]\([^ ]*\)[ ][-]\?[-]\?[ ]\?/\1 \2 -- /g'
ACCEPT icmpv6 -- ::/0 ::/0
```

**Comment by erig0** [comment]
Fixed in #1360.
